CI fails with dev dependencies: ValueError: zero-size array to reduction operation maximum which has no identity

CI fails with dev dependencies: https://github.com/huggingface/trl/actions/runs/18712033739/job/53362520844
> ValueError: zero-size array to reduction operation maximum which has no identity

```python
FAILED tests/test_online_dpo_trainer.py::TestOnlineDPOTrainer::test_training_with_transformers_paged[standard_prompt_only] - ValueError: zero-size array to reduction operation maximum which has no identity
FAILED tests/test_online_dpo_trainer.py::TestOnlineDPOTrainer::test_training_with_transformers_paged[conversational_prompt_only] - ValueError: zero-size array to reduction operation maximum which has no identity
```

Stacktrace:
```python
   >       trainer.train()
  
  tests/test_online_dpo_trainer.py:452: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  .venv/lib/python3.12/site-packages/transformers/trainer.py:2156: in train
      return inner_training_loop(
  .venv/lib/python3.12/site-packages/transformers/trainer.py:2487: in _inner_training_loop
      tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/online_dpo_trainer.py:1230: in training_step
      prompt_ids, prompt_mask, completion_ids, completion_mask = self._generate(model, prompts, images)
                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/online_dpo_trainer.py:1078: in _generate
      completion_ids = pad(completion_ids, padding_value=self.pad_token_id, padding_side="right")
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/utils.py:348: in pad
      output_shape = np.max([t.shape for t in tensors], 0).tolist()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/numpy/_core/fromnumeric.py:3164: in max
      return _wrapreduction(a, np.maximum, 'max', axis, None, out,
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  obj = [], ufunc = <ufunc 'maximum'>, method = 'max', axis = 0, dtype = None
  out = None
  kwargs = {'initial': <no value>, 'keepdims': <no value>, 'where': <no value>}
  passkwargs = {}
  
      def _wrapreduction(obj, ufunc, method, axis, dtype, out, **kwargs):
          passkwargs = {k: v for k, v in kwargs.items()
                        if v is not np._NoValue}
      
          if type(obj) is not mu.ndarray:
              try:
                  reduction = getattr(obj, method)
              except AttributeError:
                  pass
              else:
                  # This branch is needed for reductions like any which don't
                  # support a dtype.
                  if dtype is not None:
                      return reduction(axis=axis, dtype=dtype, out=out, **passkwargs)
                  else:
                      return reduction(axis=axis, out=out, **passkwargs)
      
  >       return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  E       ValueError: zero-size array to reduction operation maximum which has no identity
  
  .venv/lib/python3.12/site-packages/numpy/_core/fromnumeric.py:86: ValueError
```

Context:
- https://github.com/huggingface/trl/issues/4035
- https://github.com/huggingface/trl/pull/4046
- https://github.com/huggingface/transformers/pull/40692

--- a/trl/trainer/online_dpo_trainer.py
+++ b/trl/trainer/online_dpo_trainer.py
@@ -26,6 +26,7 @@ import torch
 import torch.nn as nn
 import torch.nn.functional as F
 import torch.utils.data
+import transformers
 from accelerate import logging
 from accelerate.utils import broadcast_object_list, gather_object, is_peft_model
 from datasets import Dataset
@@ -1049,10 +1050,11 @@ class OnlineDPOTrainer(BaseTrainer):
         if self.use_transformers_paged:
             previous_attn = self.model_wrapped.config._attn_implementation
 
-            if is_flash_attn_2_available():
-                self.model_wrapped.config._attn_implementation = "paged_attention"
+            if version.parse(transformers.__version__).release >= version.parse("5.0.0").release:
+                new_attn = "paged|flash_attention_2" if is_flash_attn_2_available() else "paged|sdpa"
             else:
-                self.model_wrapped.config._attn_implementation = "sdpa_paged"
+                new_attn = "paged_attention" if is_flash_attn_2_available() else "sdpa_paged"
+            self.model_wrapped.config._attn_implementation = new_attn
             with (
                 profiling_context(self, "transformers.generate_batch"),
                 unwrap_model_for_generation(

